src/java/org/jivesoftware/openfire/Connection.java

```diff
@@ -361,9 +337,10 @@
      * <tt>clientMode</tt> parameter since it will behave as the server in the TLS negotiation.
      *
      * @param clientMode boolean indicating if this entity is a client or a server in the TLS negotiation.
+     * @param directTLS boolean indicating if the negotiation is directTLS (true) or startTLS (false).
      * @throws Exception if an error occured while securing the connection.
      */
-    void startTLS(boolean clientMode) throws Exception;
+    void startTLS(boolean clientMode, boolean directTLS) throws Exception;
```

Comment:
Pretty sure that because of XEP-0368 mandating SNI, you'll need to be able to pass in a remote domain here, too.